### ISSUE TITLE
Cannot deselect constraint in Cluster view

### ISSUE THREAD

**Issue by anoopsarkar** [sfu-natlang/lensingwikipedia#149]
Clicking anywhere in the Cluster view creates the following constraint:

![image](https://cloud.githubusercontent.com/assets/193767/6006389/fa1414da-aac6-11e4-899b-1888b0bb1b58.png)

Clicking on the X does not get rid of this constraint.

The same problem occurs when you select some points, e.g. a Cluster constraint of 45 points and then pressing X creates the same zero event constraint as above.

**Comment by anoopsarkar** [comment]
There is an unrelated bug that after the first selection, when you make a new selection to replace the previous one, the height of the selection box is always the max of the height of the previous selection boxes.

**Comment by anoopsarkar** [comment]
For the 0 points cluster bug. Clicking on "Clear all Constraints" works.

**Comment by KonceptGeek** [comment]
@anoopsarkar creating a separate issue for selection box size bug. Have fixed the one related to constraint list in the branch - https://github.com/sfu-natlang/lensingwikipedia/tree/cluster-bug-fix
Do you want me to open a pull request for that fix and take a look at the bounding box size bug later?

**Comment by anoopsarkar** [comment]
yes, please open a pull request if this issue has been resolved.
